File: backend/Wms.Api/Domain/Entities/GoodsReceipt.cs
namespace Wms.Api.Domain.Entities;

public class GoodsReceipt
{
    public int Id { get; set; }
    public required string ReceiptNumber { get; set; }
    public required string SupplierName { get; set; }
    public int WarehouseId { get; set; }
    public int LocationId { get; set; }
    public DateTime ReceivedAtUtc { get; set; } = DateTime.UtcNow;
    public Warehouse Warehouse { get; set; } = null!;
    public Location Location { get; set; } = null!;
    public ICollection<GoodsReceiptLine> Lines { get; set; } = [];
}

File: frontend/wms-web/src/types/goodsReceipt.ts
export type GoodsReceiptLine = {
  id: number
  productId: number
  productSku: string
  productName: string
  quantity: number
}

export type GoodsReceipt = {
  id: number
  receiptNumber: string
  supplierName: string
  warehouseId: number
  purchaseOrderId: number | null
  purchaseOrderNumber: string | null
  warehouseName: string
  locationId: number
  locationCode: string
  receivedAtUtc: string
  lines: GoodsReceiptLine[]
}

export type CreateGoodsReceiptRequest = {
  supplierName: string
  warehouseId: number
  locationId: number
  purchaseOrderId: number | null
  lines: Array<{ productId: number; quantity: number }>
}
